source/gems/dry-validation/comparison-with-activemodel.html.md

```diff
@@ -0,0 +1,606 @@
+**dry-validation**
+
+```ruby
+required(:attr).confirmation
+```
+
+> Note: ActiveModel automatically creates a virtual confirmation attribute for you whose name is the name of the field that has to be confirmed with "_confirmation" appended. If you are using Protected Parameters you will need to add this attribute yourself.
+
+### 2.4 exclusion
+
+This helper validates that the attributes' values are not included in a given enumerable object.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, exclusion: { in: enumerable_object }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(exclusion?: enumerable_object)
+```
+
+> Note: As per ActiveModel docs, `:within` option is an alias of `:in`
+
+### 2.5 format
+
+This helper validates the attributes' values by testing whether they match or doesn't match a given regular expression.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, format: { with: regex }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).required(format?: regex)
+```
+
+####Doesn't Match
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, format: { without: regex }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr) { filled? & !format?(regex) }
+```
+
+### 2.6 inclusion
+
+This helper validates that the attributes' values are included in a given enumerable object.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, inclusion: { in: enumerable_object }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(inclusion?: enumerable_object)
+```
+
+> Note: As per ActiveModel docs, `:within` option is an alias of `:in`
+
+### 2.7 length
+
+This helper validates the length of the attribute's value. ActiveModel relies on a variety of options to specify length constraints in different ways. dry-validation uses different predicates for each constraint.
+
+####Minimum
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { minimum: int }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(min_size?: int)
+```
+
+####Maximum
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { maximum: int }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(max_size?: int)
+```
+
+####In
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { in: range }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(size?: range)
+```
+
+####Is
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, length: { is: int }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(size?: int)
+```
+
+#### Tokeniser Option
+
+As with ActiveModel Validations, dry-validation counts characters by default. ActiveModel provides a `:tokeniser` option to allow you to customise how the value is split. You can achieve the same thing in dry-validation by creating your own predicate, e.g.:
+
+```ruby
+Dry::Validation.Schema do
+  required(:attr).filled { word_count?(min_size: 300, max_size: 400) }
+
+  configure do
+    def word_count?(value, options={})
```

Comment:
this should be `word_count?(options, value)` - predicates always receive values as the last argument